### PR TITLE
11 get topic query2

### PR DESCRIPTION
**Pull request by sofefaal**

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -329,13 +330,24 @@ describe("GET: /api/users", () => {
...
     })
   })
+  test("GET: responds with a 404 when passed a topic query which does not exist ", () => {
+    return request(app)
+      .get("/api/articles?topic=hello")
+      .expect(404)
+      .then(({body}) => {
+        expect(body).toEqual({message: "Sorry topic not found"});
+      });
+  })
```
Comment (nc-katherine-hurst) [inline __tests__/integration.test.js:352]: This is great, should also add one more test to check that you get a 200 and an empty array if you try and get articles for an existing topic, for example `/api/articles/?topic=paper`

File: models/articles.model.js [hunk 2]
```diff
@@ -11,52 +11,61 @@ function fetchArticleId(article_id) {
...
+    const topicQuery = ` WHERE articles.topic=$1`;
+    sqlQuery += topicQuery;
+    articleTopic.push(topic);
+  }
+
+  sqlQuery += ` GROUP BY articles.article_id`;
 
+  return db.query(sqlQuery, articleTopic)
+  .then(({ rows: articles }) => {
+    if (articles.length === 0) {
```
Comment (nc-katherine-hurst) [inline models/articles.model.js:38]: Will need to change this slightly as if a topic exists but has no articles, it would need to return an empty array and a 200 rather than a 404
To overcome this, we'll need to check if a topic exists or not - remember you have an endpoint that does this 👀

**Review by nc-katherine-hurst — COMMENTED** [review]
Looks good Sofe, just need to add and pass one more test 🧪
